src/initialise.jl

```diff
@@ -108,13 +108,38 @@ function make_rational_map!(
     skyrmion,
     pfn,
     qfn,
-    prof;
+    prof = nothing;
+    baryon = nothing,
     X = [0.0, 0.0, 0.0],
     iTH = 0.0,
     i_n = [0.0, 0.0, 1.0],
     jTH = 0.0,
     j_n = [0.0, 0.0, 0.0],
+    print_things = true,
 )
+    # If a profile function is not given choose a sensible choice
+    if isnothing(prof)
+        # If the baryon number has not been specified, deduce it from from the
+        # polynomials provided
+        if isnothing(baryon)
+            baryon1 = abs((log(pfn(10000)) - log(pfn(1)))/log(10000))
+            baryon2 = abs((log(qfn(10000)) - log(qfn(1)))/log(10000))
+            baryon = max(round(baryon1), round(baryon2))
+            if print_things == true
+                println(
+                    "I think your baryon number is ",
+                    baryon,
+                    ". If it is not, include '; baryon=B' in your argument.",
+                )
+            end
+        end
+        R(z) = pfn(z)/qfn(z)
+        k1, k2 = getOKprofile(1.0, baryon, getI(R), skyrmion.mpi)
+        # Need what seems like an extra line here to avoid "cannot add method
+        # to function argument" error here. 
+        nprof(r) = pi/(1 - tanh(-k2*k1))*(-tanh(k2*(r - k1)) + 1.0);
+        prof = nprof
```

Comment:
Would a closure be neater? e.g. `prof = r -> pi/(1 - tanh(-k2*k1))*(-tanh(k2*(r - k1)) + 1.0;`, this also sidesteps the error.

Reply:
(adding to review)